pgxn/neon/communicator.c

```diff
@@ -1045,6 +1078,25 @@ prefetch_do_request(PrefetchRequest *slot, neon_request_lsns *force_request_lsns
 	Assert(!found);
 }
 
+/*
+ * Check that returned page LSN is consistent with request lsns
+ */
+static void
+check_page_lsn(NeonGetPageResponse* resp)
+{
+	if (neon_protocol_version < 3) /* no information to check */
```

Comment:
This may break if the protocol version is upgraded, as still incoming responses may not yet be of the new protocol version. Or do we not care for such cases?